src/core/src/pass/serialize.cpp

```diff
@@ -89,13 +92,32 @@ class ConstantWriter {
           m_enable_compression(enable_compression),
           m_blob_offset(bin_data.tellp()) {}
 
-    FilePosition write(const char* ptr, size_t size) {
+    FilePosition write(const char* ptr,
+                       size_t size,
+                       bool compress_to_fp16 = false,
+                       ov::element::Type src_type = ov::element::dynamic,
+                       int64_t* new_size = nullptr) {
         const FilePosition write_pos = m_binary_output.tellp();
         const auto offset = write_pos - m_blob_offset;
+        *new_size = size;
+
+        if (compress_to_fp16) {
+            // TODO: Remove this duplicate
+            if (src_type == ov::element::f32) {
+                *new_size = size / 2;
+            } else if (src_type == ov::element::f64) {
+                *new_size = size / 4;
```

Comment:
sizeof(type) instead of magic numbers?

> TODO: Remove this duplicate

what does it imply?

Reply:
Removed